### PR TITLE
Chore: Remove isEqual mock from field state test

### LINKED ISSUE #80039: Remove isEqual mock in fieldState unit test
As a part of the yarn nodeModules work, we spotted a test failure coming from lodash.isEqual being mocked out with a radically different implementation than the real thing, introduced by https://github.com/grafana/grafana/pull/75511 

This mock needs to be removed and fix the broken implementation.

### PR DESCRIPTION
**Pull request by nmarrs**
Dependent on https://github.com/grafana/grafana/pull/81323 being merged

Removes isEqual mock in test. On reviewing original code it appears that we can update the logic to be an OR between the two methods of determining whether or not two fields are "equal". Tested previous bugs ([geomap tooltip values having "1" appended](https://github.com/grafana/grafana/pull/75511) + [barchart tooltips adhering to field overrides](https://github.com/grafana/grafana/pull/67872)) and did not notice any regressions.

https://github.com/grafana/grafana/assets/22381771/dd895434-ae2a-4980-9b5d-444b3758d0e6

Fixes https://github.com/grafana/grafana/issues/80039

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

**Review by baldm0mma — APPROVED** [review]

File: packages/grafana-data/src/field/fieldState.test.ts [hunk 1]
```diff
@@ -162,14 +157,20 @@ describe('Check field state calculations (displayName and id)', () => {
     const title = checkScenario({
       frames: [
         toDataFrame({
-          fields: [{ name: TIME_SERIES_VALUE_FIELD_NAME }, { name: TIME_SERIES_VALUE_FIELD_NAME }],
+          fields: [
+            { name: TIME_SERIES_VALUE_FIELD_NAME, config: { test: 1 } },
+            { name: TIME_SERIES_VALUE_FIELD_NAME, config: { test: 2 } },
```
Comment (ryantxu) [inline packages/grafana-data/src/field/fieldState.test.ts:162]: what is the config?  something seems fishy here -- since i don't know how this is valid (if it fixes something, also seems fishy)

Comment (nmarrs) [reply packages/grafana-data/src/field/fieldState.test.ts:162]: Original `===` check was doing strict equality, meaning that the compared fields were the same object in memory while the added `isEqual` is doing deep equality where objects can be the same if they match on deep comparison.

This PR changes the logic to say that the field is the same field either if it passes strict or deep equality. Given this the tests needed to be updated to slightly modify the fields being passed so they were not passing deep equality. For a long time we were doing just strict equality, my PR a few months ago switched this to just deep comparison, and now to get rid of the lodash mock I have switched it to either or which _seems_ to not break anything.

That being said... there is a chance accepting both types of equality for determining if the field is the same may have some strange edge cases 😬

Comment (nmarrs) [reply packages/grafana-data/src/field/fieldState.test.ts]: all of this was cleaned up by https://github.com/grafana/grafana/pull/81323
